lib/yaml/representer.py

```diff
@@ -44,35 +57,67 @@ def represent_data(self, data):
             #self.represented_objects[alias_key] = None
             self.object_keeper.append(data)
         data_types = type(data).__mro__
-        if data_types[0] in self.yaml_representers:
-            node = self.yaml_representers[data_types[0]](self, data)
+        yaml_representers = self.yaml_representers()
+        yaml_multi_representers = self.yaml_multi_representers()
+        if data_types[0] in yaml_representers:
+            node = yaml_representers[data_types[0]](self, data)
         else:
             for data_type in data_types:
-                if data_type in self.yaml_multi_representers:
-                    node = self.yaml_multi_representers[data_type](self, data)
+                if data_type in yaml_multi_representers:
+                    node = yaml_multi_representers[data_type](self, data)
                     break
             else:
-                if None in self.yaml_multi_representers:
-                    node = self.yaml_multi_representers[None](self, data)
-                elif None in self.yaml_representers:
-                    node = self.yaml_representers[None](self, data)
+                if None in yaml_multi_representers:
+                    node = yaml_multi_representers[None](self, data)
+                elif None in yaml_representers:
+                    node = yaml_representers[None](self, data)
                 else:
                     node = ScalarNode(None, str(data))
         #if alias_key is not None:
         #    self.represented_objects[alias_key] = node
         return node
 
+    @classmethod
+    def yaml_representers(cls):
+        cls.representer_setup.ensure_initialized()
+
+        if cls.representer_registry.yaml_representers_initialized:
+            return cls.representer_registry.yaml_representers
+
+        constructor_cls = next(c for c in cls.mro() if hasattr(c, 'yaml_representers') and c is not cls)
+        return constructor_cls.yaml_representers()
+
+    @classmethod
+    def yaml_multi_representers(cls):
+        cls.representer_setup.ensure_initialized()
+
+        if cls.representer_registry.yaml_multi_representers_initialized:
+            return cls.representer_registry.yaml_multi_representers
```

Comment:
I think you should be able to do the quick return if it's initialized *before* calling `ensure_initialized`, also anywhere else this pattern happens in this PR, I didn't search carefully.